src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -8,46 +8,78 @@
 import edu.wpi.first.math.geometry.Rotation2d;
 import edu.wpi.first.math.geometry.Translation2d;
 import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.networktables.NetworkTableEvent;
+import edu.wpi.first.networktables.NetworkTableInstance;
 import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.DriverStation.Alliance;
 import edu.wpi.first.wpilibj.Filesystem;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.team2412.robot.Robot;
+import frc.team2412.robot.Robot.RobotType;
 import java.io.File;
+import java.util.EnumSet;
 import java.util.function.DoubleSupplier;
 import java.util.function.Supplier;
 import swervelib.SwerveDrive;
 import swervelib.math.SwerveMath;
 import swervelib.parser.SwerveParser;
+import swervelib.telemetry.SwerveDriveTelemetry;
+import swervelib.telemetry.SwerveDriveTelemetry.TelemetryVerbosity;
 
 public class DrivebaseSubsystem extends SubsystemBase {
 
 	// SWERVE CONSTANTS (that aren't in deploy dir)
 
-	private static final double MAX_SPEED = 2.0;
+	private static final double MAX_SPEED = 4.4196;
 	private static final double JOYSTICK_DEADBAND = 0.05;
+	// TODO: change this to allow for comp robot
 	private static final double DRIVEBASE_RADIUS =
-			Math.hypot(8.5, 8.5); // our wheels are 8.5 inches by 8.5 inches from the center of the bot;
+			Robot.getInstance().getRobotType() == RobotType.PRACTICE
+					? Math.hypot(
+							8.5, 8.5) // our wheels are 8.5 inches by 8.5 inches from the center of the bot;
+					: 0.3937; // crane radius
+	private static final double HEADING_CORRECTION_DEADBAND = 0.005;
 
 	// AUTO CONSTANTS
 
+	public static final Rotation2d MAX_ROTATIONS_PER_SEC = Rotation2d.fromRotations(0.8574);
+
 	private static final PIDConstants AUTO_TRANSLATION_PID = new PIDConstants(0.1, 0, 0);
 	private static final PIDConstants AUTO_ROTATION_PID = new PIDConstants(5.0, 0, 0);
 	private static final double MAX_AUTO_SPEED =
 			500.0; // this seems to only affect rotation for some reason
 
 	private final SwerveDrive swerveDrive;
+	private final ShuffleboardTab drivebaseTab = Shuffleboard.getTab("Drivebase");
+
+	private boolean xWheelsEnabled = true;
+
+	// shuffleboard variables
+	private GenericEntry headingCorrectionEntry;
```

Comment:
Should be `final` so the compiler checks it was initialized during class construction